pkg/injection/namespace/mapper/dynakubes.go

```diff
@@ -20,32 +22,63 @@ type DynakubeMapper struct {
 	dk         *dynakube.DynaKube
 	operatorNs string
 	secrets    k8ssecret.QueryObject
+
+	matchedOANamespaces []string
+	matchedMENamespaces []string
 }
 
 func NewDynakubeMapper(ctx context.Context, clt client.Client, apiReader client.Reader, operatorNs string, dk *dynakube.DynaKube) DynakubeMapper {
 	return DynakubeMapper{
-		ctx:        ctx,
-		client:     clt,
-		apiReader:  apiReader,
-		operatorNs: operatorNs,
-		dk:         dk,
-		secrets:    k8ssecret.Query(clt, apiReader, log),
+		ctx:                 ctx,
+		client:              clt,
+		apiReader:           apiReader,
+		operatorNs:          operatorNs,
+		dk:                  dk,
+		secrets:             k8ssecret.Query(clt, apiReader, log),
+		matchedOANamespaces: []string{},
+		matchedMENamespaces: []string{},
 	}
 }
 
 // MapFromDynakube checks all the namespaces to all the dynakubes
 // updates the labels on the namespaces if necessary,
 // finds conflicting dynakubes (2 dynakube with codeModules on the same namespace)
-func (dm DynakubeMapper) MapFromDynakube() error {
+func (dm *DynakubeMapper) MapFromDynakube() error {
 	modifiedNs, err := dm.MatchingNamespaces()
 	if err != nil {
-		return errors.Cause(err)
+		return err
+	}
+
+	if err := dm.updateNamespaces(modifiedNs); err != nil {
+		return err
+	}
+
+	if dm.dk.OneAgent().IsAppInjectionNeeded() {
+		log.Info("namespaces monitored",
+			"selector", "OneAgent",
+			"count (at most 10 are displayed)", len(dm.matchedOANamespaces),
+			"namespaces", dm.matchedOANamespaces,
+		)
 	}
 
-	return dm.updateNamespaces(modifiedNs)
+	if dm.dk.MetadataEnrichment().IsEnabled() {
+		log.Info("namespaces monitored",
+			"selector", "MetadataEnrichment",
+			"count (at most 10 are displayed)", len(dm.matchedMENamespaces),
+			"namespaces", dm.matchedMENamespaces,
+		)
+	}
+
+	oaActive := dm.dk.OneAgent().IsAppInjectionNeeded()
+	meActive := dm.dk.MetadataEnrichment().IsEnabled()
+	setNamespacesMonitoredSelectorCondition(dm.dk.Conditions(), "OneAgent", oaActive, dm.matchedOANamespaces)
+	setNamespacesMonitoredSelectorCondition(dm.dk.Conditions(), "MetadataEnrichment", meActive, dm.matchedMENamespaces)
```

Comment:
why do you have extra `if`s here for the logs?

you could log the same thing in `setNamespacesMonitoredSelectorCondition` as you have all the info